src/pages/ca-adjustment/form/CAadjustmentForms.js

```diff
@@ -0,0 +1,319 @@
+                error={formik.touched.reference && Boolean(formik.errors.reference)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={SystemRepository.Plant.qry}
+                name='plantId'
+                label={labels.plant}
+                valueField='recordId'
+                displayField={['reference', 'name']}
+                columnsInDropDown={[
+                  { key: 'reference', value: 'plant Ref' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                values={formik.values}
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  const plantId = newValue?.recordId || ''
+                  formik.setFieldValue('plantId', plantId)
+                }}
+                error={formik.touched.plantId && Boolean(formik.errors.plantId)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <CustomDatePicker
+                name='date'
+                label={labels.date}
+                value={formik.values.date}
+                onChange={formik.setFieldValue}
+                required
+                maxAccess={maxAccess}
+                onClear={() => formik.setFieldValue('date', '')}
+                error={formik.touched.date && Boolean(formik.errors.date)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={SystemRepository.Currency.qry}
+                name='currencyId'
+                label={labels.currency}
+                valueField='recordId'
+                displayField={['reference', 'name']}
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                values={formik.values}
+                required
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('currencyId', newValue?.recordId || null)
+                }}
+                error={formik.touched.currencyId && Boolean(formik.errors.currencyId)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <ResourceLookup
+                endpointId={CashBankRepository.CashAccount.snapshot}
+                parameters={{
+                  _type: 2
+                }}
+                valueField='reference'
+                displayField='name'
+                name='cashAccountId'
+                required
+                label={labels.cashAccount}
+                form={formik}
+                valueShow='cashAccountRef'
+                secondValueShow='cashAccountName'
+                onChange={(event, newValue) => {
+                  if (newValue) {
+                    formik.setFieldValue('cashAccountId', newValue?.recordId)
```

Comment:
use ternary operator for less code pls
example:
formik.setFieldValue('cashAccountId',newValue? newValue.recordId : null)

Reply:
there is a bug after using this this code we try it and we were using setvalues also